src/chunk.rs

```diff
@@ -0,0 +1,105 @@
+use std::iter::*;
+use std::fmt::Debug;
+
+/// Chunk of code text
+#[derive(Debug)]
+pub enum Chunk {
+    Text(String),
+    EndLine,
+}
```

Comment:
Chunk's purpose to separate unchunkable part of string.
Ex. for `    fn add<T: Upcast<Widget>>(&self, widget: &T) {`
will be these chunks `Indent(1)`, `Text("fn add<T: Upcast<Widget>>(")`, `Text("&self,")`,
`Separator`, `Text("widget: &T")`, `Text(") {")`
Formater will split long result lines by `Separator`.
For line width 40, `Text("widget: &T")` will go to new string with `Indent(2)` instead of `Separator` before him.
You know better solution?